Makefile

```diff
@@ -1,139 +1,360 @@
+
+.PHONY: docker-verify
+docker-verify: ## to verify the docker image
+	@$(INFO) Verifying the published docker image...
+	$(AT)echo "$${COSIGN_PUBLIC_KEY}" > cosign_public.key && \
+	$(DOCKER) run ${DOCKER_OPTS} \
+	--entrypoint '/bin/sh' \
+	-v $(PWD):/app -w /app \
+	${DOCKER_IMAGE_COSIGN} \
+	-c \
+	"echo Verifying... && \
+	cosign verify --key cosign_public.key $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}:${APP_VERSION}" || ${FAIL}
+# if we are on a latest semver APP_VERSION tag, also verify latest tag
+ifneq ($(shell echo $(APP_VERSION) | egrep '^v([0-9]+\.){0,2}(\*|[0-9]+)'),)
+  ifeq ($(shell git tag -l --sort=v:refname | tail -n1),$(APP_VERSION))
+	$(DOCKER) run ${DOCKER_OPTS} \
+	--entrypoint '/bin/sh' \
+	-v $(PWD):/app -w /app \
+	${DOCKER_IMAGE_COSIGN} \
+	-c \
+	"echo Verifying... && \
+	cosign verify --key cosign_public.key $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}:latest" || ${FAIL}
+  endif
+endif
+	$(AT)rm -f cosign_public.key || ${FAIL}
+	@$(OK) Verifying the published docker image: $(DOCKER_REGISTRY)/${DOCKER_REGISTRY_REPO}:${APP_VERSION}
+
+.PHONY: docker-sbom
+docker-sbom: ## to print a sbom report
+	@$(INFO) Performing Docker sbom report...
+	$(AT)$(DOCKER) sbom ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker sbom report
+
+.PHONY: docker-scan
+docker-scan: ## to print a vulnerability report
+	@$(INFO) Performing Docker scan report...
+	$(AT)$(DOCKER) scan ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker scan report
+
+.PHONY: docker-lint
+docker-lint: ## to lint the Dockerfile
+	@$(INFO) Dockerfile linting...
+	$(AT)$(DOCKER) run -i ${DOCKER_OPTS} \
+	${DOCKER_IMAGE_DOCKERLINT} \
+	< ${DOCKER_FILE} || ${FAIL}
+	@$(OK) Dockerfile linting
 
 .PHONY: docker-login
-## docker-login: docker login to registry
-docker-login:
-	@echo Docker login
-	echo $(DOCKER_PASSWORD) | docker login --username $(DOCKER_USERNAME) --password-stdin
+docker-login: ## to login to a container registry
+	@$(INFO) Dockerd login to container registry ${DOCKER_REGISTRY}...
+	$(AT) echo "${DOCKER_PASSWORD}" | $(DOCKER) login --password-stdin -u ${DOCKER_USER} $(DOCKER_REGISTRY) || ${FAIL}
+	@$(OK) Dockerd login to container registry ${DOCKER_REGISTRY}...
 
-.PHONY: docker-push
-## docker-push: push the docker image
-docker-push: docker-login
-	@echo Taging Pushing Ponos Serve Docker Image
-	docker push mattermost/mattermost-ponos-service:${TAG}
-
-.PHONY: check-modules
-## check-modules: Check outdated modules
-check-modules:
-	@echo Checking outdated modules
-	$(GO) list -u -m -json all | docker run --rm -i psampaz/go-mod-outdated
-
-.PHONY: check-style
-## check-style: Runs govet and gofmt against all packages.
-check-style: govet lint
-	@echo Checking for style guide compliance
+go-build: $(GO_BUILD_PLATFORMS_ARTIFACTS) ## to build binaries
 
-.PHONY: clean
-## clean: deletes all
-clean:
-	rm -rf build/_output/bin/
-
-.PHONY: dist
-## dist-aws: creates the bundle file for AWS Lambda deployments
-dist: build
```

Comment:
Are we sure that lambda deployment bundle is not needed ?

Reply:
good catch @pfltdv  👍 

Will make sure to include it in the templated makefile.

Reply:
I was ready to say that this is definitely needed

Reply:
Pushed https://github.com/mattermost/ponos/pull/2/commits/364bd1a0f1b2f63d2170b661be68147aa0172ec6 to assess this